source/MicroBitMemoryMap.cpp

```diff
@@ -138,5 +142,64 @@ void MicroBitMemoryMap::findHashes()
                 memoryMapStore.memoryMap[2].startAddress = (uint32_t)magicAddress;
                 return;
         }
+        
+        if (   *(magicAddress - (sizeof(magicAddress))) == 0x597F30FE
+            && *(magicAddress - (sizeof(magicAddress)/4))   == 0xC1B1D79D
+           ) {
+            // Found uPy Magic
+            DMESG("FOUND UPY_MAGIC");
+            uint8_t nRegions = (*(magicAddress - (sizeof(magicAddress)/2)));
+            
+            // Iterate through regions
+            for(int i = 0; i < nRegions; i++) {
+               processRecord(magicAddress - ((2 + i) * sizeof(magicAddress))); 
```

Comment:
I think this will break if another region was added to the table.

Reply:
Yeah you're right as the struct has a set size. I'm not sure what the best way to avoid this is, if we allocate more now or later when we need it

From `inc/MicroBitMemoryMap.h`
```
#define NUMBER_OF_REGIONS 3
    struct MemoryMapStore
    {
        Region memoryMap[NUMBER_OF_REGIONS];
    };
```

Reply:
Oh! I didn't spot that you are using nRegions. I was just thinking the code uses the last few regions in the table. I had better check my code! I think it will be OK if processRecord ignores any new record.id's that don't make an index that fits in the array.

Reply:
👍